nsxt/resource_nsxt_policy_l7_access_profile_test.go

```diff
@@ -0,0 +1,369 @@
+			},
+			{
+				ResourceName:      testResourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccResourceNsxtPolicyL7AccessProfile_importBasic_multitenancy(t *testing.T) {
+	name := getAccTestResourceName()
+	testResourceName := "nsxt_policy_l7_access_profile.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+			testAccOnlyMultitenancy(t)
+		},
+		Providers: testAccProviders,
+		CheckDestroy: func(state *terraform.State) error {
+			return testAccNsxtPolicyL7AccessProfileCheckDestroy(state, name)
+		},
+		Steps: []resource.TestStep{
+			{
+				Config: testAccNsxtPolicyL7AccessProfileMinimalistic(true),
+			},
+			{
+				ResourceName:      testResourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateIdFunc: testAccResourceNsxtPolicyImportIDRetriever(testResourceName),
+			},
+		},
+	})
+}
+
+func testAccNsxtPolicyL7AccessProfileExists(displayName string, resourceName string) resource.TestCheckFunc {
+	return func(state *terraform.State) error {
+
+		connector := getPolicyConnector(testAccProvider.Meta().(nsxtClients))
+
+		rs, ok := state.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Policy L7AccessProfile resource %s not found in resources", resourceName)
+		}
+
+		resourceID := rs.Primary.ID
+		if resourceID == "" {
+			return fmt.Errorf("Policy L7AccessProfile resource ID not set in resources")
+		}
+
+		exists, err := resourceNsxtPolicyL7AccessProfileExists(testAccGetSessionContext(), resourceID, connector)
+		if err != nil {
+			return err
+		}
+		if !exists {
+			return fmt.Errorf("Policy L7AccessProfile %s does not exist", resourceID)
+		}
+
+		return nil
+	}
+}
+
+func testAccNsxtPolicyL7AccessProfileCheckDestroy(state *terraform.State, displayName string) error {
+	connector := getPolicyConnector(testAccProvider.Meta().(nsxtClients))
+	for _, rs := range state.RootModule().Resources {
+
+		if rs.Type != "nsxt_policy_l7_access_profile" {
+			continue
+		}
+
+		resourceID := rs.Primary.Attributes["id"]
+		exists, err := resourceNsxtPolicyL7AccessProfileExists(testAccGetSessionContext(), resourceID, connector)
+		if err == nil {
+			return err
+		}
+
+		if exists {
+			return fmt.Errorf("Policy L7AccessProfile %s still exists", displayName)
+		}
+	}
+	return nil
+}
+
+func testAccNsxtPolicyL7AccessProfileTemplate(createFlow bool, withContext bool) string {
+	var attrMap map[string]string
+	if createFlow {
+		attrMap = accTestPolicyL7AccessProfileCreateAttributes
+	} else {
+		attrMap = accTestPolicyL7AccessProfileUpdateAttributes
+	}
+	context := ""
+	if withContext {
+		context = testAccNsxtPolicyMultitenancyContext()
+	}
+	return fmt.Sprintf(`
+resource "nsxt_policy_l7_access_profile" "test" {
+%s
+  display_name = "%s"
+  description  = "%s"
+  default_action_logged = %s
+  default_action = "%s"
+
+  l7_access_entry {
+    nsx_id = "%s"
```

Comment:
for better coverage we need tests without nsx_id, and ideally with a mix (some entries with id, some without)